File: app/src/main/java/adapter/TrailerAdapter.java
package adapter;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.google.android.youtube.player.YouTubeInitializationResult;
import com.google.android.youtube.player.YouTubeThumbnailLoader;
import com.google.android.youtube.player.YouTubeThumbnailView;
import com.google.firebase.analytics.FirebaseAnalytics;
import com.google.firebase.crash.FirebaseCrash;

import java.util.List;

import activity.TrailerActivity;
import br.com.icaro.filme.R;
import info.movito.themoviedbapi.model.Video;
import utils.Config;
import utils.Constantes;

/**
 * Created by icaro on 22/02/17.
 */
public class TrailerAdapter extends RecyclerView.Adapter<TrailerAdapter.TrailerViewHolder> {
    Context context;
    private List<Video> videos;
    private String sinopse;

    public TrailerAdapter(FragmentActivity activity, List<Video> videos, String overview) {
        context = activity;
        this.videos = videos;
        this.sinopse = overview;
    }

    @Override
    public TrailerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.scroll_trailer, parent, false);
        return new TrailerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(TrailerAdapter.TrailerViewHolder holder, final int position) {
        final String youtube_key = videos.get(position).getKey();

        holder.play_view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(context, TrailerActivity.class);
                //  Log.d("OnClick", youtube_key);
                intent.putExtra(Constantes.YOU_TUBE_KEY, youtube_key);
                intent.putExtra(Constantes.SINOPSE, sinopse);

                context.startActivity(intent);

                Bundle bundle = new Bundle();
                bundle.putString(FirebaseAnalytics.Event.SELECT_CONTENT, TrailerActivity.class.getName());
                bundle.putString("Endereço do youtube", youtube_key);
                FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.SELECT_CONTENT, bundle);
            }
        });

        holder.thumbnailView.initialize(Config.YOUTUBE_API_KEY, OnInitializedListener(youtube_key));

    }


    @Override
    public int getItemCount() {
        return videos.size();
    }

    private YouTubeThumbnailView.OnInitializedListener OnInitializedListener(final String youtube_key) {
        return new YouTubeThumbnailView.OnInitializedListener() {

            @Override
            public void onInitializationSuccess(YouTubeThumbnailView youTubeThumbnailView, YouTubeThumbnailLoader youTubeThumbnailLoader) {
                youTubeThumbnailLoader.setVideo(youtube_key);
            }

            @Override
            public void onInitializationFailure(YouTubeThumbnailView youTubeThumbnailView, YouTubeInitializationResult youTubeInitializationResult) {
                FirebaseCrash.report(new Exception("Erro em \"onInitializationFailure\" dentro de " + this.getClass()));
            }
        };
    }


    public class TrailerViewHolder extends RecyclerView.ViewHolder {
        FrameLayout play_view;
        YouTubeThumbnailView thumbnailView;

        public TrailerViewHolder(View itemView) {
            super(itemView);
            play_view = (FrameLayout) itemView.findViewById(R.id.frame_youtube_view_thumbnail);
            thumbnailView = (YouTubeThumbnailView) itemView.findViewById(R.id.youtube_view_thumbnail);

        }
    }
}

File: app/src/main/java/adapter/TrailerAdapter.kt
package adapter

import adapter.TrailerAdapter.*
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.*
import br.com.icaro.filme.R
import br.com.icaro.filme.R.*
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubeThumbnailLoader
import com.google.android.youtube.player.YouTubeThumbnailView
import com.google.android.youtube.player.YouTubeThumbnailView.*
import com.google.firebase.crashlytics.FirebaseCrashlytics
import domain.ResultsVideosItem
import trailer.TrailerActivity
import utils.Constant
import utils.key.ApiKeys
import utils.visible

/**
 * Created by icaro on 22/02/17.
 */
class TrailerAdapter(private val videos: MutableList<ResultsVideosItem?>?, private val sinopse: String) : Adapter<TrailerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = TrailerViewHolder(parent = parent)
    override fun onBindViewHolder(holder: TrailerViewHolder, position: Int) = holder.bind(videos?.get(position)!!)
    override fun getItemCount() = videos?.size ?: 0

    inner class TrailerViewHolder(parent: ViewGroup)
        : RecyclerView.ViewHolder(LayoutInflater.from(parent.context)
        .inflate(layout.scroll_trailer, parent, false)) {
        private val YOUTUBE_KEY = ApiKeys.YOUTUBE_API_KEY

        fun bind(video: ResultsVideosItem) = with(itemView)  {
            contentDescription = video.name
            findViewById<YouTubeThumbnailView>(R.id.youtube_view_thumbnail)
                .initialize(YOUTUBE_KEY, object : OnInitializedListener {
                    override fun onInitializationSuccess(
                        youTubeThumbnailView: YouTubeThumbnailView,
                        youTubeThumbnailLoader: YouTubeThumbnailLoader
                    ) {
                        findViewById<ImageView>(R.id.play_treiler_img).visible()
                        youTubeThumbnailLoader.setVideo(video.key)
                    }

                    override fun onInitializationFailure(
                        youTubeThumbnailView: YouTubeThumbnailView,
                        youTubeInitializationResult: YouTubeInitializationResult
                    ) {
                        FirebaseCrashlytics.getInstance().log("Erro em \"onInitializationFailure\" dentro de " + this.javaClass)
                    }
                })
            setOnClickListener {
                context.startActivity(Intent(context, TrailerActivity::class.java).apply {
                    putExtra(Constant.YOU_TUBE_KEY, video.key)
                    putExtra(Constant.SINOPSE, sinopse)
                })
            }
        }
    }

    fun addVideos(videosTrailers: MutableList<ResultsVideosItem?>?) {
            videos?.addAll(videosTrailers?.toList() ?: listOf())
            notifyDataSetChanged()
    }

}